features/steps/steps.go

```
@@ -2,87 +2,204 @@ package steps
-	// ctx.Step(`^I should receive a hello-world response$`, c.iShouldReceiveAHelloworldResponse)
+	ctx.Step(`^the following instance with id "([^"]*)" is available from dp-dataset-api:$`, c.theFollowingInstanceIsAvailable)
+	ctx.Step(`^the instance with id "([^"]*)" has ([^"]*) dimension options`, c.theInstanceHasDimensionCount)
+	ctx.Step(`^the following response is available from Cantabular from the codebook "([^"]*)" and query "([^"]*)":$`, c.theFollowingCodebookIsAvailable)
+
+	ctx.Step(`^the call to add a dimension to the instance with id "([^"]*)" is successful`, c.theCallToAddInstanceDimensionIsSuccessful)
+	ctx.Step(`^the instance with id "([^"]*)" is successfully updated`, c.theCallToUpdateInstanceIsSuccessful)
+	ctx.Step(`^the job with id "([^"]*)" is successfully updated`, c.theCallToUpdateJobIsSuccessful)
+
+	ctx.Step(`^this category-dimension-import event is consumed:$`, c.thisCategoryDimensionImportEventIsConsumed)
+	ctx.Step(`^these instance-complete events are produced:$`, c.theseDimensionCompleteEventsAreProduced)
+}
+
+// theFollowingInstanceIsAvailable generate a mocked response for dataset API
+// GET /instances/{id} with the provided instance response
+func (c *Component) theFollowingInstanceIsAvailable(id string, instance *godog.DocString) error {
+	c.DatasetAPI.NewHandler().
+		Get("/instances/"+id).
+		Reply(http.StatusOK).
+		BodyString(instance.Content).
+		AddHeader("Etag", testETag)
+
+	return nil
 }
 
-func (c *Component) iShouldReceiveAHelloworldResponse() error {
-	content, err := ioutil.ReadFile(c.cfg.OutputFilePath)
+// theInstanceHasDimensionCount generates a mocked response for dataset API
+// GET /instances/{id}/dimensions with the provided totalCount and limit=offset=0
+func (c *Component) theInstanceHasDimensionCount(id string, totalCount string) error {
+
+	// count must be a number
+	cnt, err := strconv.Atoi(totalCount)
 	if err != nil {
-		return err
+		return nil
```

Why don't you need to return the error here?

---

I don't know, I think this should be an error, I will change it to return the error.